precompiles/staking/integration_test.go

```diff
@@ -406,6 +406,94 @@ var _ = Describe("Calling staking precompile directly", func() {
+				Website:         "[do-not-modify]",
+				SecurityContact: "[do-not-modify]",
+				Details:         "[do-not-modify]",
+			}
+			defaultCommissionRate    = big.NewInt(staking.DoNotModifyCommissionRate)
+			defaultMinSelfDelegation = big.NewInt(staking.DoNotModifyMinSelfDelegation)
+
+			// defaultEditValidatorArgs are the default arguments for the editValidator call
+			//
+			// NOTE: this has to be populated in the BeforeEach block because the private key is not initialized before
+			defaultEditValidatorArgs contracts.CallArgs
+		)
+
+		BeforeEach(func() {
+			// populate the default editValidator args
+			defaultEditValidatorArgs = defaultCallArgs.WithMethodName(staking.EditValidatorMethod)
+		})
+
+		Context("validator address is the origin", func() {
+			It("should edit validator success", func() {
+				// create a new validator
+				newAddr, newPriv := testutiltx.NewAccAddressAndKey()
+				hexAddr := common.BytesToAddress(newAddr.Bytes())
+				err := evmosutil.FundAccountWithBaseDenom(s.ctx, s.app.BankKeeper, newAddr, 2e18)
+				Expect(err).To(BeNil(), "error while funding account: %v", err)
+
+				description := staking.Description{
+					Moniker:         "new node",
+					Identity:        "",
+					Website:         "",
+					SecurityContact: "",
+					Details:         "",
+				}
+				commission := staking.Commission{
+					Rate:          big.NewInt(100000000000000000),
+					MaxRate:       big.NewInt(100000000000000000),
+					MaxChangeRate: big.NewInt(100000000000000000),
+				}
+				minSelfDelegation := big.NewInt(1)
+				pubkeyBase64Str := "UuhHQmkUh2cPBA6Rg4ei0M2B04cVYGNn/F8SAUsYIb4="
+				value := big.NewInt(1e18)
+
+				createValidatorArgs := defaultCallArgs.WithMethodName(staking.CreateValidatorMethod).
+					WithPrivKey(newPriv).
+					WithArgs(description, commission, minSelfDelegation, hexAddr, pubkeyBase64Str, value)
+
+				logCheckArgs := passCheck.WithExpEvents(staking.EventTypeCreateValidator)
+
+				_, _, err = contracts.CallContractAndCheckLogs(s.ctx, s.app, createValidatorArgs, logCheckArgs)
+				Expect(err).To(BeNil(), "error while calling the smart contract: %v", err)
+
+				s.NextBlock()
+
+				// edit validator
+				editValidatorArgs := defaultEditValidatorArgs.
+					WithPrivKey(newPriv).
+					WithArgs(defaultDescription, hexAddr, defaultCommissionRate, defaultMinSelfDelegation)
+
+				logCheckArgs = passCheck.WithExpEvents(staking.EventTypeEditValidator)
+
+				_, _, err = contracts.CallContractAndCheckLogs(s.ctx, s.app, editValidatorArgs, logCheckArgs)
+				Expect(err).To(BeNil(), "error while calling the smart contract: %v", err)
+
+				validator, found := s.app.StakingKeeper.GetValidator(s.ctx, newAddr.Bytes())
+				Expect(found).To(BeTrue(), "expected validator to be found")
+				Expect(validator.Description.Moniker).To(Equal(defaultDescription.Moniker), "expected validator moniker is updated")
+			})
+		})
+
+		Context("validator address is not the origin", func() {
```

Comment:
Nit: can you please add a test case that the user updates a validator using a smart contract that calls the precompile?

Reply:
Something similar. With the current logic, a smart contract address could not be used as validator operator on the create/edit validator transaction. But a user could use a smart contract that calls the precompile to edit a validator using the user's address. That's the test case I'm referring to

Reply:
I agree with @GAtom22 for completeness we should have the corresponding smart contract test.

You can extend the `StakingCaller.sol` contract to include a method `editValidator` which just calls the staking precompile's method with the given arguments. Then create a corresponding logic in the smart contract "branch" of the integration tests:

```go
var _ = Describe("Calling staking precompile via Solidity", func() {
    // ...
    Context("to edit a validator", func() {
        // add test cases here
    })
})
```

Reply:
added 2 test cases for this